### PR TITLE
Toggle the filter tree for the trace overview

### LINKED ISSUE #787: Make the filter tree togglable or hidden for the trace overview.
The trace overview should have its filter tree togglable or hidden since its main purpose is to provide context for the whole trace.

### PR DESCRIPTION
**Pull request by hoangphamEclipse**
This commit creates a hamburger menu for the trace overview that will allow users to toggle the filter tree and select a different output view. The hamburger menu replaces the existing cog icon that is used for output source selection. A new state property is added to the AbstractTreeOutputComponent that allows showing and hiding the filter tree. The overview then toggles this state property.

Fix #787

### CODE REVIEW

File: packages/react-components/src/components/abstract-tree-output-component.tsx
Comment (marco-miller) [inline packages/react-components/src/components/abstract-tree-output-component.tsx:23]: Extra blank line introduced, would remove it.

File: packages/react-components/src/components/abstract-tree-output-component.tsx
Comment (marco-miller) [inline packages/react-components/src/components/abstract-tree-output-component.tsx:48]: Extra blank line?

**Review by marco-miller — CHANGES_REQUESTED** [review]
LGTM, with just a few nits.

**Review by marco-miller — DISMISSED** [review]
Letting you @hoangphamEclipse decide if you need a second review; thanks, let me/us know.

**Review by Rodrigoplp-work — CHANGES_REQUESTED** [review]
Awesome!
Could you add the name of the selected source to the title of the overview (the text that shows on the vertical bar on the left of the chart)?
So, after the user clicks on the hamburger button, clicks on "Select source output" and chooses for example "CPU Usage", the title would change from "Overview" to "Overview - CPU Usage".
If that is too much work we can leave it to be done on another ticket, as this was not specified in the requirements.

**Comment by hoangphamEclipse** [comment]
> Awesome! Could you add the name of the selected source to the title of the overview (the text that shows on the vertical bar on the left of the chart)? So, after the user clicks on the hamburger button, clicks on "Select source output" and chooses for example "CPU Usage", the title would change from "Overview" to "Overview - CPU Usage". If that is too much work we can leave it to be done on another ticket, as this was not specified in the requirements.

This can be done in another PR.

**Review by Rodrigoplp-work — APPROVED** [review]

**Review by marco-miller — APPROVED** [review]
LGTM still, letting you @Rodrigoplp-work rebase/merge at will, if you tested this latest update. I didn't, but I tested an earlier version (of this PR).

**Pull request merged**
